### PR TITLE
Remove a lot of inline event handlers

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
Remove a lot of inline event handlers as part of the ongoing Content-Security-Policy work.

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: api/src/org/labkey/api/util/element/Input.java [hunk 1]
```diff
@@ -580,12 +594,15 @@ protected void doStyles(Appendable sb) throws IOException
         }
     }
 
-    protected void doInputEvents(Appendable sb) throws IOException
+    protected void doInputEvents(String id, Appendable sb) throws IOException
```
Comment (labkey-tchad) [inline api/src/org/labkey/api/util/element/Input.java]: Can we remove the unused `Appendable` parameter?
```suggestion
    protected void doInputEvents(String id) throws IOException
```

File: core/src/org/labkey/core/view/template/bootstrap/ViewServiceImpl.java [hunk 2]
```diff
@@ -189,24 +188,21 @@ public void doStartTag(PrintWriter out)
...
                 String expandCollapseGifId = "expandCollapse-" + rootId;
 
                 boolean isCollapsible = config._isCollapsible;
                 if (isCollapsible)
                 {
                     if (rootId == null)
                         isCollapsible = false;
-                    else
-                        expandCollapseUrl = PageFlowUtil.urlProvider(ProjectUrls.class).getExpandCollapseURL(context.getContainer(), "", rootId);
                 }
```
Comment (labkey-tchad) [inline core/src/org/labkey/core/view/template/bootstrap/ViewServiceImpl.java:198]: Is the plan to get rid of `ProjectUrls.getExpandCollapseURL` entirely? It does seem weird to have an `href` on a expand/collapse link.

File: query/src/org/labkey/query/controllers/QueryController.java [hunk 3]
```diff
@@ -6903,8 +6903,11 @@ public ModelAndView getView(Object o, BindException errors)
             List<QueryParseException> qpe = new ArrayList<>();
             String expr = getViewContext().getRequest().getParameter("q");
             ArrayList<String> html = new ArrayList<>();
-            html.add("<form method=GET><textarea id=\"expression\" cols=100 rows=10 name=q>" + PageFlowUtil.filter(expr) + "</textarea><br><input type=submit onclick='Ext.getBody().mask();'></form>\n" +
-                    "<script>" +
+            PageConfig config = getPageConfig();
+            var inputId = config.makeId("submit_");
+            config.addHandler(inputId, "click", "Ext.getBody().mask();");
+            html.add("<form method=GET><textarea id=\"expression\" cols=100 rows=10 name=q>" + PageFlowUtil.filter(expr) + "</textarea><br><input id=\"" + inputId + "\" type=submit></form>\n" +
+                    "<script type=\"application/javascript\" nonce=\"" + config.getScriptNonce() + "\">" +
```
Comment (labkey-tchad) [inline query/src/org/labkey/query/controllers/QueryController.java]:
```suggestion
                    "<script type=\"text/javascript\" nonce=\"" + config.getScriptNonce() + "\">" +

```

File: core/src/org/labkey/core/view/template/bootstrap/ViewServiceImpl.java [hunk 4]
```diff
@@ -488,8 +485,9 @@ public void renderPortalMenu(PrintWriter out, String title)
...
                 if (PageFlowUtil.isPageAdminMode(getContext()))
                 {
-                    NavTree permissionsNav = new NavTree("Permissions",
-                            "javascript:LABKEY.Portal._showPermissions(" +
+                    NavTree permissionsNav = new NavTree("Permissions");
+                    permissionsNav.setScript(
+                            "LABKEY.Portal._showPermissions(" +
                                     config._webpart.getRowId() + "," +
                                     permissionString + "," +
                                     containerPathString + ");"
```
Comment (labkey-tchad) [inline core/src/org/labkey/core/view/template/bootstrap/ViewServiceImpl.java]: I think this needs to return false. Clicking this adds to the browser history as it is.
```suggestion
                    permissionsNav.setScript(
                            "LABKEY.Portal._showPermissions(" +
                                    config._webpart.getRowId() + "," +
                                    permissionString + "," +
                                    containerPathString + "); return false;"
```

File: issues/src/org/labkey/issue/view/detailView.jsp [hunk 5]
```diff
@@ -202,17 +202,21 @@
                 }
                 if (!getUser().isGuest())
                 {
-                    navTree.addChild("Create related " + names.singularName.toLowerCase(), relatedIssues.toString());
+                    navTree.addChild("Create related " + names.singularName.toLowerCase()).setScript(relatedIssues.toString());
```
Comment (labkey-tchad) [inline issues/src/org/labkey/issue/view/detailView.jsp]: Script should return false. (Not sure where the correct place for the change should be)
```suggestion
                    navTree.addChild("Create related " + names.singularName.toLowerCase()).setScript(relatedIssues.toString() + "; return false;");

```

File: query/src/org/labkey/query/DataViewsWebPartFactory.java [hunk 6]
```diff
@@ -116,8 +116,8 @@ public WebPartView getWebPartView(@NotNull ViewContext portalCtx, @NotNull Porta
                     view.setCustomize(customize);
                 }
 
-                String editScript = "editDataViews(" + webPart.getRowId() + ");";
-                NavTree edit = new NavTree("Edit", "javascript:" + editScript, null, "fa fa-pencil");
+                NavTree edit = new NavTree("Edit", "#editDataViews", null, "fa fa-pencil");
+                edit.setScript("editDataViews(" + webPart.getRowId() + ");");
```
Comment (labkey-tchad) [inline query/src/org/labkey/query/DataViewsWebPartFactory.java]: Return false?
```suggestion
                NavTree edit = new NavTree("Edit", "#editDataViews", null, "fa fa-pencil");
                edit.setScript("editDataViews(" + webPart.getRowId() + "); return false;");

```

File: study/src/org/labkey/study/controllers/reports/ReportsController.java [hunk 7]
```diff
@@ -1269,8 +1269,8 @@ public ModelAndView getView(ParticipantReportForm form, BindException errors)
                 view.setTitle(StudyService.get().getSubjectNounSingular(getContainer()) + " Report");
                 view.setFrame(WebPartView.FrameType.PORTAL);
 
-                String script = String.format("javascript:customizeParticipantReport('%s');", form.getComponentId());
-                NavTree edit = new NavTree("Edit", script, null, "fa fa-pencil");
+                NavTree edit = new NavTree("Edit", null, null, "fa fa-pencil");
+                edit.setScript(String.format("customizeParticipantReport('%s');", form.getComponentId()));
```
Comment (labkey-tchad) [inline study/src/org/labkey/study/controllers/reports/ReportsController.java]:
```suggestion
                edit.setScript(String.format("customizeParticipantReport('%s'); return false;", form.getComponentId()));

```

File: study/src/org/labkey/study/reports/ParticipantReport.java [hunk 8]
```diff
@@ -95,8 +95,8 @@ public HttpView renderReport(ViewContext context)
 
         if (canEdit(context.getUser(), context.getContainer()))
         {
-            String script = String.format("javascript:customizeParticipantReport('%s');", form.getComponentId());
-            NavTree edit = new NavTree("Edit", script, null, "fa fa-pencil");
+            NavTree edit = new NavTree("Edit", null, null, "fa fa-pencil");
+            edit.setScript(String.format("customizeParticipantReport('%s');", form.getComponentId()));
```
Comment (labkey-tchad) [inline study/src/org/labkey/study/reports/ParticipantReport.java]:
```suggestion
            edit.setScript(String.format("customizeParticipantReport('%s'); return false;", form.getComponentId()));

```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
Found a few scripts passed to `NavTree.setScript` that should return `false` (I think).

File: issues/src/org/labkey/issue/view/detailView.jsp (hunk 5 above)
Comment (labkey-matthewb) [reply issues/src/org/labkey/issue/view/detailView.jsp]: NavTree is odd because you don't actually know how it will render.  In any event, it never hurts.

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**